datalad/core/local/run.py

```diff
@@ -454,24 +454,16 @@ def format_command(dset, command, **kwds):
 
 
 def _execute_command(command, pwd, expected_exit=None):
-    from datalad.cmd import Runner
+    from datalad.cmd import WitlessRunner
 
     exc = None
     cmd_exitcode = None
-    runner = Runner(cwd=pwd)
+    runner = WitlessRunner(cwd=pwd)
     try:
         lgr.info("== Command start (output follows) =====")
         runner.run(
-            command,
-            # immediate output
-            log_online=True,
-            # not yet sure what we should do with the command output
-            # IMHO `run` itself should be very silent and let the command talk
-            log_stdout=False,
-            log_stderr=False,
-            expect_stderr=True,
-            expect_fail=True,
-            # TODO stdin
+            # command is always a string
+            ['sh', '-c', command],
```

Comment:
This is a problem, and will not work on windows properly (or at least be fragile). While there is a `sh` it may behave totally different than the actual calling shell. For example, with a system-wide installed git, executed through an `sh.exe` process inside a normal `cmd.exe` terminal, it might not read/find the user's `.gitconfig` (thanks to @adswa for pointing that out).

Here is a demo:

With `sh -c`:

```
(base) C:\DLTMP\some>datalad run git config --show-origin -l
file:C:/Program Files/Git/etc/gitconfig diff.astextplain.textconv=astextplain
file:C:/Program Files/Git/etc/gitconfig filter.lfs.clean=git-lfs clean -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.smudge=git-lfs smudge -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.process=git-lfs filter-process
file:C:/Program Files/Git/etc/gitconfig filter.lfs.required=true
file:C:/Program Files/Git/etc/gitconfig http.sslbackend=openssl
file:C:/Program Files/Git/etc/gitconfig http.sslcainfo=C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt
file:C:/Program Files/Git/etc/gitconfig core.autocrlf=true
file:C:/Program Files/Git/etc/gitconfig core.fscache=true
file:C:/Program Files/Git/etc/gitconfig core.symlinks=false
file:C:/Program Files/Git/etc/gitconfig pull.rebase=false
file:C:/Program Files/Git/etc/gitconfig credential.https://dev.azure.com.usehttppath=true
file:C:/Program Files/Git/etc/gitconfig credential.helper=manager-core
file:.git/config        core.repositoryformatversion=0
file:.git/config        core.filemode=false
file:.git/config        core.bare=false
file:.git/config        core.logallrefupdates=true
file:.git/config        core.symlinks=false
file:.git/config        core.ignorecase=true
file:.git/config        annex.uuid=7bc9391d-e0eb-4ce0-bb16-10ecc2cefdc9
file:.git/config        annex.sshcaching=false
file:.git/config        annex.crippledfilesystem=true
file:.git/config        annex.version=8
file:.git/config        annex.backends=MD5E
file:.git/config        filter.annex.smudge=git-annex smudge -- %f
file:.git/config        filter.annex.clean=git-annex smudge --clean -- %f
```

and without:

```
(base) C:\DLTMP\some>datalad run git config --show-origin -l
file:C:/Program Files/Git/etc/gitconfig diff.astextplain.textconv=astextplain
file:C:/Program Files/Git/etc/gitconfig filter.lfs.clean=git-lfs clean -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.smudge=git-lfs smudge -- %f
file:C:/Program Files/Git/etc/gitconfig filter.lfs.process=git-lfs filter-process
file:C:/Program Files/Git/etc/gitconfig filter.lfs.required=true
file:C:/Program Files/Git/etc/gitconfig http.sslbackend=openssl
file:C:/Program Files/Git/etc/gitconfig http.sslcainfo=C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt
file:C:/Program Files/Git/etc/gitconfig core.autocrlf=true
file:C:/Program Files/Git/etc/gitconfig core.fscache=true
file:C:/Program Files/Git/etc/gitconfig core.symlinks=false
file:C:/Program Files/Git/etc/gitconfig pull.rebase=false
file:C:/Program Files/Git/etc/gitconfig credential.https://dev.azure.com.usehttppath=true
file:C:/Program Files/Git/etc/gitconfig credential.helper=manager-core
file:C:/Users/mih/.gitconfig    user.email=michael.hanke@gmail.com
file:C:/Users/mih/.gitconfig    user.name=Michael Hanke
file:C:/Users/mih/.gitconfig    annex.security.allowed-url-schemes=http https file local
file:C:/Users/mih/.gitconfig    annex.security.allowed-url-addresses=all
file:.git/config        core.repositoryformatversion=0
file:.git/config        core.filemode=false
file:.git/config        core.bare=false
file:.git/config        core.logallrefupdates=true
file:.git/config        core.symlinks=false
file:.git/config        core.ignorecase=true
file:.git/config        annex.uuid=7bc9391d-e0eb-4ce0-bb16-10ecc2cefdc9
file:.git/config        annex.sshcaching=false
file:.git/config        annex.crippledfilesystem=true
file:.git/config        annex.version=8
file:.git/config        annex.backends=MD5E
file:.git/config        filter.annex.smudge=git-annex smudge -- %f
file:.git/config        filter.annex.clean=git-annex smudge --clean -- %f
```

Any use of `sh -c` in the tests is likely to run into the same problem.

Reply:
Ha, and this might also explain the magic file name mangling that is happening on windows with this PR.

When we send a windows path to `sh -c` it will never work. The reason for that is that it will come in as python string `'C:\\something\\else'`, which will translate to the command `sh -c "C:\something\else"`, and `sh` will apply standard handling of `\` as an escape char, and kill all "path separators".

Overall all these quirks seem to suggest that we should only very carefully, probably never, try to apply UNIXy paradigm to windows. It might work for a few cases and some layers, but then it breaks.